core/src/Kokkos_Macros.hpp

```diff
@@ -58,6 +58,13 @@
 #include <impl/Kokkos_NvidiaGpuArchitectures.hpp>
 #endif
 
+#include <iso646.h>
```

Comment:
The C++20 way would be using `version` but that might not exist for problematic toolchains.

Reply:
Any good reason to prefer the version you proposed to
```C++
#ifndef KOKKOS_ENABLE_CXX17
#include <version>
#if defined(_GLIBCXX_RELEASE) && _GLIBCXX_RELEASE < 10
#error \
    "Compiling with support for C++20 or later requires a libstdc++ version later than 9"
#endif
#endif

Reply:
`version` might not exist and we would get a better error message with the current code. I'm fine with splitting the if in two and only conditionally include `iso646.h`.

Reply:
```C++
#if __has_include(<version>)
#include <version>
#else
#include <ciso646>
#endif
```
then?

Reply:
Sure, just in case there is a case where `iso646.h` doesn't define `_GLIBCXX_RELEASE` but `<version>` exists.